Sources/BitcoinKitPrivate/BitcoinKit.Private.swift

```diff
@@ -225,23 +223,25 @@ public class _HDKey {
             defer {
                 BN_free(n)
             }
-            var result = [UInt8](repeating: 0, count: 33)
             EC_POINT_point2bn(group, point, POINT_CONVERSION_COMPRESSED, n, ctx)
-            BN_bn2bin(n, &result)
-            let fingerprintData = _Hash.sha256ripemd160(publicKey)
-            let fingerprintArray = fingerprintData.withUnsafeBytes {
-                [UInt32](UnsafeBufferPointer(start: $0, count: fingerprintData.count))
+            result = Data(count: 33)
+            result.withUnsafeMutableBytes { (ptr: UnsafeMutablePointer<UInt8>) in
+                BN_bn2bin(n, ptr)
+                return
             }
-            let reusltData = Data(result)
-            return _HDKey(privateKey: reusltData,
-                          publicKey: reusltData,
-                          chainCode: derivedChainCode,
-                          depth: depth + 1,
-                          fingerprint: fingerprintArray[0],
-                          childIndex: childIndex)
-        } else {
-            return nil
         }
+        
+        let fingerprintData = _Hash.sha256ripemd160(publicKey)
+        let fingerprint = fingerprintData.withUnsafeBytes{ (p: UnsafePointer<UInt32>) in
+            p.pointee
+        }
+        return _HDKey(privateKey: result,
+                      publicKey: result,
```

Comment:
Are really both of `privateKey` and `publicKey` the same value?

Reply:
Oh this is kind of bad architecture of _HDKey.
I think _HDKey should be refactored, but it's not in the scope of this PR.